### DISCUSSION TITLE
Error after launch_fluent execution

### DISCUSSION THREAD

**Discussion by robcodesnew** [ansys/pyfluent#2279] [Q&A]
Hello. I am launching fluent in meshing mode with a journal sucessfully. The journal runs and my mesh is valid. Upon completion, I get this error. This simply stops me from running the rest of my simulation automatically. Is there a way to solve it easily so that I do not have to manually interfere after the meshing phase?
![image](https://github.com/ansys/pyfluent/assets/150433762/c1542969-0417-4418-b840-9c0ab3e75794)

**Comment by raph-luc** [comment]
@robcodesnew I haven't seen that issue before, but Fluent 2022 R1 is not supported, see the [readme](https://github.com/ansys/pyfluent): "PyFluent supports Fluent 2022 R2 and later."

Can you try again with a more recent Ansys Fluent release?

**Reply by robcodesnew** [reply]
Thanks for the reply @raph-luc. 
I seem to struggle to launch a more recent version, despite having 22R2 installed. If I run the simple command `pyfluent.launch_fluent(show_gui=True, product_version="22.2.0")`,a window in 22R1 opens. I have tried manually opening 22R2 and it works just fine. My environment variables seem correct to me (see image). Is there something I'm missing?

![image](https://github.com/ansys/pyfluent/assets/150433762/f445c5fc-27ae-450a-a05e-d1a890ddee03)


Thanks

**Reply by raph-luc** [reply]
It is difficult for me to tell without having access to your system. 

Can you try setting the env var `PYFLUENT_FLUENT_ROOT=C:\Program Files\ANSYS Inc\v222\fluent`?

Make sure that the environment variables in Command Prompt or whatever console you are using are also up-to-date. They can differ from the ones you set in your system. Opening a new console (or even rebooting your machine) will typically synchronize them.

If you set `PYFLUENT_FLUENT_ROOT`, you should not need to specify the Fluent version when launching it through PyFluent, just `pyfluent.launch_fluent()` should launch the version specified in `PYFLUENT_FLUENT_ROOT`.

**Reply by robcodesnew** [reply]
@raph-luc Thanks, that did fix my issues. I am now able to run fluent 22R2. Unfortunately, that did not fix the initial error. If you do not know what it is, I will investigate on my end. However, I would happily take any suggestions if you have any.

**Reply by raph-luc** [reply]
Is the error message still the exact same as the one you posted above? 

What console are you using? Can you try using Command Prompt? What PyFluent version are you using?

**Reply by robcodesnew** [reply]
I am launching fluent from a command in a jupyter notebook running on a venv. Here are my versions. I will try from the command prompt
![image](https://github.com/ansys/pyfluent/assets/150433762/5a9ac633-d85c-472b-8592-5bcba6645da0)

**Reply by raph-luc** [reply]
Please try with the latest released version, which is `pip install ansys-fluent-core==0.18.2` (it is newer than `0.19.dev1` which was just a development tag)

**Reply by raph-luc** [reply]
In addition, how exactly are you using PyFluent to launch Fluent? Can you share the script you are using? Are you specifying any paths for the PyFluent commands?

It seems that your system is using shortened DOS 8.3 paths (such as using `C:\PROGRA~2` instead of `C:\Program Files (x86)`, see the second line in your screenshot), this could be causing these issues. Which Windows version are you using?

**Reply by robcodesnew** [reply]
This is my code to launch fluent. 
`working_dir = tmp_filepath # Current working directory for fluent`  
`journal_paths ="-i " + tmp_filepath + geom_name + "_aoa_" + str(aoa_list[0])+ "_journal_pre.jou "`    
`pyfluent.launch_fluent(version = "3d",
                        mode="meshing", 
                        precision = "single", 
                        processor_count = 12, 
                        additional_arguments = journal_paths, 
                        cwd = working_dir,
                        show_gui=True,
                        )`

**Reply by robcodesnew** [reply]
I am using windows 10

**Reply by raph-luc** [reply]
Can you check if one of those paths that you are providing to PyFluent are in the DOS 8.3 format? Just do `print(working_dir)` and `print(journal_paths )`, you don't need to share them here

**Reply by robcodesnew** [reply]
It seems that the error is due to the the way I input my journals, (LaunchFluentError: 
Fluent Launch string: start "" "C:\Program Files\ANSYS Inc\v222\fluent\ntbin\win64\fluent.exe" 3d -t12 -meshing -i [PATH/tmp/naca2412_aoa_10.0_journal_pre.jou](file:///PATH/naca2412_aoa_10.0_journal_pre.jou)  -sifile="C:\PATH\Local\Temp\serverinfo-iu97nppv.txt" -nm)

This was recommended to me in [#2230 ](https://github.com/ansys/pyfluent/discussions/2230#discussioncomment-7558284)

**Reply by robcodesnew** [reply]
Both filepaths are not in DOS 8.3 formats

**Reply by raph-luc** [reply]
OK, and if you don't specify the journals then everything works fine? 

By the way, you should never get any of those `pyfluent... WARNING` or `pyfluent... ERROR` messages that you showed in your screenshot (before the traceback). If any of those are happening, there is something wrong.

Just so we can rule out any more basic/system-related issues, can you try:
``` 
import ansys.fluent.core as pyfluent
session = pyfluent.launch_fluent()
print(session.health_check_service.check_health())
session.exit()
```
You should get no error messages or warnings, output should look like this:
![image](https://github.com/ansys/pyfluent/assets/132297401/9bd24970-6710-4c0c-a3b0-d41f092e1deb)

**Reply by robcodesnew** [reply]
![image](https://github.com/ansys/pyfluent/assets/150433762/bb9158b7-156f-4a8f-a424-dc00909e33cc)
Seems ok!

**Reply by raph-luc** [reply]
That is good. And can you confirm whether you are specifying the path using forward or backwards slashes? The command you shared was:
`-i PATH/tmp/naca2412_aoa_10.0_journal_pre.jou`, but Windows typically uses backward slashes.

Can you try specifying paths relative to where Fluent is launched rather than absolute paths? Perhaps move the journal file to Fluent's working directory so you can specify only the journal file name rather than the full path?

FYI, the next PyFluent release is going to have better support for journal files name specification like the ones you are trying to do here, so these issues should not happen anymore

**Reply by robcodesnew** [reply]
The journal path is already in fluent's wd. Do you recommend I just type -i journal.jou ?

**Reply by raph-luc** [reply]
Correct, that should work in my understanding

**Reply by robcodesnew** [reply]
I have run this and I unfortunately get the same error. Fluent is indeed launched and I get the results but still the same message after the meshing. 
________
The above exception was the direct cause of the following exception:

LaunchFluentError                         Traceback (most recent call last)
[z:\folder\folder\folder\file.ipynb](file:///z:\folder\folder\folder\file.ipynb) Cell 21 line 8
      [5](vscode-notebook-cell:/z%3A/\folder\folder\folder\file.ipynb.ipynb#X26sZmlsZQ%3D%3D?line=4) journal_paths ="-i " + geom_name + "_aoa_10.0_journal_pre.jou " 
      [7](vscode-notebook-cell:/z%3A/\folder\folder\folder\file.ipynb.ipynb#X26sZmlsZQ%3D%3D?line=6) #Launch fluent with the journal
...
    [780](file:///Z:/folders/pyfluentvenv/Lib/site-packages/ansys/fluent/core/launcher/launcher.py:780) finally:
    [781](file:///Z:/folders/pyfluentvenv/Lib/site-packages/ansys/fluent/core/launcher/launcher.py:781)     server_info_file = Path(server_info_filepath)
    
    
    
   I will unfortunatey have to go offline for 30 mins, but I really appreciate the help

**Reply by raph-luc** [reply]
@robcodesnew Can you please try this quick test: [test.zip](https://github.com/ansys/pyfluent/files/13531444/test.zip)
Extract this into an empty folder, open a Command Prompt, navigate to the folder where you extracted this, and execute `test.bat`. 

This will create a clean virtual environment, install PyFluent, and run the Python `test.py` file. which will print out the working dir, start Fluent 2022 R2, and set Fluent to read the `test/test.jou` file, which only initializes the WTM workflow and then exits. 

The only output that you should see (after the PyFluent install) is the working directory printed out, and the `SERVING` message.

**Reply by robcodesnew** [reply]
![image](https://github.com/ansys/pyfluent/assets/150433762/f4f44a32-a3fb-42ea-b556-5860eb801461)
All good! Thanks for the test

**Reply by raph-luc** [reply]
OK, so this shows that there is most likely something wrong/unexpected inside that journal file that you are passing to Fluent, or there is some issue related to how you were launching Fluent previously, perhaps with the paths/working directories that you were using. 

Try to work starting from the files that I provided, and figure out what change that you made that breaks execution.

**Reply by robcodesnew** [reply]
The only trouble I have explaining this to be the issue is my journals are in a chain (i.e jorunal 1 has "read journal 2"at the end of it and so on) and the error only appears at the end of the cycle, not for every journal (my journals are all the same except they study different angles). If I open the first journal of the chain with the GUI, I don't get an issue all the way through. I willl however take the time to start from scratch with what you have given me. I sincerely appreciate the help

**Reply by raph-luc** [reply]
No problem at all, sorry that I can't help more.

What command are you using to read subsequent journals? Something like `/file/read-journal "test2.jou"`? I just tested and this also works fine on my end. 

There has to be something wrong with your journal files, while playing around with this I noticed that if some commands inside the journal files are set incorrectly and fail to execute, then you can get all kinds of weird errors in PyFluent (not unlike the ones you shared in the original post).

**Reply by robcodesnew** [reply]
@raph-luc can you confirm what the command is to close fluent in meshing mode? I have tried running my chain with empty journals (the last of which is simply the screenshot. This exit method works for solver mode but is it different in meshing? It seems to work fine when I use it in the GUI but it gives me the error we have been talking about when I run it.
![image](https://github.com/ansys/pyfluent/assets/150433762/06567e97-567b-4468-9981-a411db0cfa98)

**Reply by raph-luc** [reply]
@robcodesnew did you figure this out? Sorry, it seems that I missed the notification from your last message. `/exit y` should indeed exit out of meshing. I tested an isolated journal file with just 
```
/file/set-tui-version "22.2"
/exit y
```
as well as a chain of journal files that ends with this one, and both still work fine with Fluent 2022 R2 on my end. 

Note that running `.jou` journal files that end with `/exit y` may raise irrelevant errors in PyFluent (e.g. `RuntimeError: failed to connect to all addresses`) at the end of execution because PyFluent never gets to properly communicate with Fluent before it exits (as Fluent is always busy working through the journal file). Ideally you should cleanly shut down Fluent from PyFluent with `session.exit()` (assuming it was launched by PyFluent with `cleanup_on_exit=True`, which is the default).
